### PR TITLE
Themes: Fix screenshot photon parameters

### PR DESCRIPTION
**Pull request by sirreal**
The `Theme` component is loading full-size screenshots, loading unnecessarily large images.

The images are meant to be resized, but a query parameter is naively added resulting in an incorrect query: `?ssl=1?w=340` which parses to `{ ssl: "1?w=340" }`.

This PR adds tests to detect the issue and uses the https://github.com/Automattic/photon.js lib to correctly generate the parameters. The final commit is optional, but rather than use a hardcoded `width`, it proposes `fit` with the largest fit I have been able to observe by viewing `https://wordpress.com/themes` with device width `479px`.

See the [photon docs](https://developer.wordpress.com/docs/photon/api/).

## Questions

* Is `fit` appropriate for this case?
* Are theme screenshot dimensions reliable?
* Is `479x360` an appropriate size?

### Before
big-thumbnail

### After
better

## Testing
* Tests should pass
* Visit https://calypso.live/themes?branch=fix/theme/bad-photon-url-params and ensure that images have appropriate quality but are no larger than expected (`479 x 360`).

Found while testing #18726

### CODE REVIEW

File: client/components/theme/test/index.jsx [hunk 1]
```diff
@@ -6,20 +6,22 @@
...
 
 /**
  * Internal dependencies
  */
 import { Theme } from '../';
 
-jest.mock( 'components/popover/menu', () => require( 'components/empty-component' ) );
-jest.mock( 'components/popover/menu-item', () => require( 'components/empty-component' ) );
+jest.mock( 'components/popover/menu', () => 'components--popover--menu' );
+jest.mock( 'components/popover/menu-item', () => 'components--popover--menu-item' );
```
Comment (ockham) [inline client/components/theme/test/index.jsx]: What kind of sorcery is this? 😮

**Comment by ockham** [comment]
> ## Questions

> * Is `fit` appropriate for this case?

Looks sensible, judging by https://developer.wordpress.com/docs/photon/api/#fit:

> Fit an image to a containing box of **w**idth,**h**eight dimensions. Image aspect ratio is maintained.

> * Are theme screenshot dimensions reliable?

I think so, but @folletto might know better.

> * Is `479x360` an appropriate size?

Deferring to Davide again 😄

File: client/components/theme/test/index.jsx (hunk 1 above)
Comment (sirreal) [reply client/components/theme/test/index.jsx]: I'm not sure whether this should remain or be reverted actually.

I did this thinking about the snapshot, but these components are not included in the shallow render so they're irrelevant.

This is a nice trick for mocks included in snapshots and doesn't harm anything here (apparently). See #19037 and #19100 for extended discussion.

TL;DR is that React will happily inject these into the DOM as elements: `<components--popover--menu any="props">Any children…</components--popover--menu>`, which is quite descriptive when compared to the render of `empty-component` (`<div />`).

**Comment by sirreal** [comment]
A note that it's important to rebuild and test logged out to hit SSR views and verify correct behavior.

**Comment by folletto** [comment]
> Are theme screenshot dimensions reliable?
> Is 479x360 an appropriate size?

I'm not sure what you mean by "reliable", but the change seems appropriate to me. The screen is also using `flex`, so it doesn't have at any moment (excluding sheer luck) a resolution that is going to be pixel perfect. So anything larger than the max size (about 310px) should be ok, if I'm not missing anything relevant. :D 

I did also a brief size comparison for the images loaded (in Chrome, using Network Inspector), and it seems we change from 2.9Mb to 935Kb. Nice.

**Comment by sirreal** [comment]
I'm seeing an issue that looks related that I hadn't seen while testing, I'll report after further investigation:

errors

**Comment by samouri** [comment]
great work @sirreal !
